### PR TITLE
Move Arguments into their own package

### PR DESCRIPTION
**Pull request by stevenschlansker**
(requires #387)

### CODE REVIEW

File: core/src/main/java/org/jdbi/v3/ArgumentRegistry.java [hunk 1]
```diff
@@ -20,6 +20,7 @@
 import java.util.Optional;
 import java.util.concurrent.CopyOnWriteArrayList;
 
+import org.jdbi.v3.argument.BuiltInArgumentFactory;
 import org.jdbi.v3.tweak.Argument;
 import org.jdbi.v3.tweak.ArgumentFactory;
```
Comment (qualidafial) [inline core/src/main/java/org/jdbi/v3/ArgumentRegistry.java]: `Argument` and `ArgumentFactory` want to move too?

I'd actually like to see the `tweak` package broken out and a lot of classes in `org.jdbi.v3` moved into subpackages by feature.

File: core/src/main/java/org/jdbi/v3/argument/SqlTypeArgument.java [hunk 2]
```diff
@@ -11,19 +11,20 @@
...
+package org.jdbi.v3.argument;
 
 import java.sql.PreparedStatement;
 import java.sql.SQLException;
 
+import org.jdbi.v3.StatementContext;
 import org.jdbi.v3.tweak.Argument;
 
-class SqlTypeArgument implements Argument
+public class SqlTypeArgument implements Argument
```
Comment (qualidafial) [inline core/src/main/java/org/jdbi/v3/argument/SqlTypeArgument.java]: Seems like this class and `ObjectArgument` are covering the same ground. Am I missing something?

File: core/src/test/java/org/jdbi/v3/argument/TestArgumentRegistry.java [hunk 3]
```diff
@@ -37,78 +40,79 @@
...
+    private final StatementContext ctx = JdbiAccess.createContext(handle);
 
     @Mock
     public PreparedStatement stmt;
 
     @Test
     public void testWaffleLong() throws Exception
     {
-        argumentRegistry.findArgumentFor(Object.class, 3L, null).get().apply(1, stmt, null);
+        ctx.findArgumentFor(Object.class, 3L).get().apply(1, stmt, null);
```
Comment (qualidafial) [inline core/src/test/java/org/jdbi/v3/argument/TestArgumentRegistry.java]: 👍

**Comment by qualidafial** [comment]
The mapping move is showing up in this PR--if you could resolve the merge conflicts that should clean up the diff for review.

**Comment by stevenschlansker** [comment]
yeah, needed to rebase, better now?

**Comment by qualidafial** [comment]
:shipit:

File: core/src/main/java/org/jdbi/v3/argument/SqlTypeArgument.java (hunk 2 above)
Comment (stevenschlansker) [inline core/src/main/java/org/jdbi/v3/argument/SqlTypeArgument.java]: they are close but not identical -- folded them together

File: core/src/main/java/org/jdbi/v3/ArgumentRegistry.java (hunk 1 above)
Comment (stevenschlansker) [inline core/src/main/java/org/jdbi/v3/ArgumentRegistry.java]: done

**Pull request merged**